src/test/java/com/mongodb/jdbc/MongoSQLResultSetTest.java

```diff
@@ -285,6 +320,175 @@ void testGetStringValues() throws Exception {
                 mongoSQLResultSet.getUnicodeStream(STRING_COL_LABEL).read(new byte[100], 0, 100));
     }
 
+    @Test
+    public void testGetStringAllTypes() throws Exception {
+        // non-null types
+        assertEquals(ALL_DOUBLE_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_DOUBLE_COL_LABEL));
+        assertEquals(ALL_OBJECT_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_OBJECT_COL_LABEL));
+        assertEquals(ALL_ARRAY_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_ARRAY_COL_LABEL));
+        assertEquals(ALL_BINARY_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_BINARY_COL_LABEL));
+        assertEquals(
+                ALL_OBJECT_ID_COL_VAL,
+                mongoSQLResultSetAllTypes.getString(ALL_OBJECT_ID_COL_LABEL));
+        assertEquals(ALL_BOOL_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_BOOL_COL_LABEL));
+        assertEquals(ALL_DATE_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_DATE_COL_LABEL));
+        assertEquals(ALL_REGEX_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_REGEX_COL_LABEL));
+        assertEquals(
+                ALL_JAVASCRIPT_COL_VAL,
+                mongoSQLResultSetAllTypes.getString(ALL_JAVASCRIPT_COL_LABEL));
+        assertEquals(ALL_SYMBOL_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_SYMBOL_COL_LABEL));
+        assertEquals(
+                ALL_JAVASCRIPT_WITH_SCOPE_COL_VAL,
+                mongoSQLResultSetAllTypes.getString(ALL_JAVASCRIPT_WITH_SCOPE_COL_LABEL));
+        assertEquals(ALL_INT_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_INT_COL_LABEL));
+        assertEquals(
+                ALL_TIMESTAMP_COL_VAL,
+                mongoSQLResultSetAllTypes.getString(ALL_TIMESTAMP_COL_LABEL));
+        assertEquals(
+                ALL_DECIMAL_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_DECIMAL_COL_LABEL));
+        assertEquals(
+                ALL_MIN_KEY_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_MIN_KEY_COL_LABEL));
+        assertEquals(
+                ALL_MAX_KEY_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_MAX_KEY_COL_LABEL));
+
+        // Note that for "string" and "long", the ALL_X_COL_VAL must use extended json style-syntax in the
```

Comment:
Using this code in the Mock will let use both canonical and relaxed extended Json represention :
Document d = Document.parse(_extendedJson_);
BsonDocument bsonDoc =  d.toBsonDocument(Document.class, MongoClient.getDefaultCodecRegistry());

For example, the snippet above accepts {"val": 1} and {"val": {"$numberInt": "1"}} as the _extendedJson_ value.

Reply:
discussed offline: I updated `MongoSQLBsonValue` to stringify all BsonValues using relaxed extended json. We will discuss in standup whether this is the behavior we want for bson strings, which will now be delimited with `"` characters.